File: Fenicia.Module.Basic/Domains/Order/Responses/GetOrderByIdResponse.cs
using Fenicia.Common.Enums.Basic;

namespace Fenicia.Module.Basic.Domains.Order.Responses;

/// <summary>
///     Response containing detailed order information including all items.
/// </summary>
public record GetOrderByIdResponse(
    Guid Id,
    string OrderNumber,
    Guid UserId,
    Guid CustomerId,
    string CustomerName,
    decimal TotalAmount,
    decimal DiscountAmount,
    int TotalQuantity,
    DateTime SaleDate,
    string Status,
    PaymentMethod PaymentMethod,
    string? Notes,
    Guid? EmployeeId = null);

File: fenicia-web/src/types/basic/order/get-order-by-id-response.ts
import { OrderDetailResponse } from "./order-detail-response";
import { OrderStatus } from "./order-status";
import { PaymentMethod } from "./payment-method";

export type GetOrderByIdResponse = {
    id: string;
    orderNumber: string;
    userId: string;
    customerId: string;
    customerName: string;
    totalAmount: number;
    discountAmount: number;
    totalQuantity: number;
    saleDate: string;
    status: OrderStatus;
    paymentMethod: PaymentMethod;
    notes?: string;
    details: OrderDetailResponse[];
    employeeId?: string;
};
